### DISCUSSION TITLE
Is it possible to have a `next/image`-based image `src` in an `og:image` tag in `head.js`?

### DISCUSSION THREAD

**Discussion by mpereira** [vercel/next.js#46263] [Help]
Or in the upcoming [SEO functionality](https://beta.nextjs.org/docs/guides/seo)?

**Comment by ciruz** [comment]
With the SEO & Metadata, you can define og tags as ``openGraph``.

Checkout the code:
https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/metadata-interface.ts#L271

And here is a list of all opengraph-types:
https://github.com/vercel/next.js/blob/5792533781e6d6fdf10d5b400d8e97d25bd5edc3/packages/next/src/lib/metadata/types/opengraph-types.ts

edit: but it's not possible to define anything other than a string or url as og:image

**Comment by mpereira** [comment]
Hey @ciruz, thanks for the suggestion.

I just tried it here, and while `generateMetadata` does inject head tags, it does not convert the given image URLs into something like what `next/image` would have for its resulting `<img>` `src` or `srcSet`. 

I'm using the latest stable Next.js, 13.1.6.

**Reply by ciruz** [reply]
Oh sorry, I read your question wrong. No this is not possible.

What are you trying to achieve? Why can't you give the same url to the og:image tag, why does it need to be the same as the image src ? because of the width/quality information in the url? or because of the srcset?

**Reply by mpereira** [reply]
> Oh sorry, I read your question wrong. No this is not possible.

No worries!

> What are you trying to achieve? Why can't you give the same url to the og:image tag, why does it need to be the same as the image src ? because of the width/quality information in the url? or because of the srcset?

For [all of the reasons](https://vercel.com/docs/concepts/image-optimization) I use `next/image` elsewhere, but mostly because `next/image` images are optimized and cached at the edge. Right now images I use in head tags are coming from my own server.

**Reply by ciruz** [reply]
Ok I gotcha now.

I honestly wouldn't invest time here, as far as I know Facebook and other services cache this information for weeks or even months, maybe forever, since you have to manually revalidate them with the FB URL Linter tool.

If someone on Facebook or for ex. Whatsapp looks at a shared link from your website, with your og:image, then Facebook doesn't fetch the picture directly from your website/server again, and again - for every facebook user. It's always coming from their cache. Maybe after some days or weeks, the original shared post has already been deleted, so the picture on facebook.com would not loaded at all. Or the website server with the og:image has a high load at the moment, and the picture would therefore be loaded extremely delayed for the facebook user. But that's never the case, because it is always cached on their servers.

One time we even had a very hard time, to change an og:image on a existing website, that has already been shared with another og:image on Whatsapp. It was always the same "old" og:image for everybody in Whatsapp, Whatsapp totally ignored the new image.

**Reply by mpereira** [reply]
Yeah that's fair. Thanks for the answer.
